examples/shader/Shader.cpp

```diff
@@ -135,15 +135,20 @@ class StormBlink : public Effect
 
     bool onLoad() override
     {
+        std::mt19937 rng = std::mt19937(std::random_device()());
```

Comment:
A few things here:

- The random device should be initialized once at the beginning of the program. I guess a global/singleton would be OK in this case...

The code can also be simplified a bit:

```suggestion
        std::mt19937 rng(global_random_device());
```

Reply:
I was running into the most vexing parse which forced me to use ` = `. I would normally just use `auto` but I wanted to match SFML style. Otherwise you're right I need to stop reinitializing that generator over and over again.

Reply:
I made that `mt19937` a global so that it's state persists across calls to that function. Not sure why I didn't just do that to begin with. For what it's worth, I'm aware that `random_device()` doesn't fully seed a `mt19937`. Doing that looks like this:
```cpp
static auto rng = []() {
    auto seed_data = std::array<int, std::mt19937::state_size>();
    auto rd = std::random_device();
    std::generate_n(seed_data.data(), seed_data.size(), std::ref(rd));
    auto seed_seq = std::seed_seq(seed_data.begin(), seed_data.end());
    return std::mt19937(seed_seq);
}();
```
I figured that all this complexity wasn't worth it for an example though.